src/modules/logger/logger.cpp

```diff
@@ -492,8 +492,7 @@ void Logger::run()
 						 */
 						message_data_header_s *header = reinterpret_cast<message_data_header_s *>(buffer);
 						header->msg_type = static_cast<uint8_t>(MessageType::DATA);
-						/* the ORB topic data object has 2 unused trailing bytes? */
-						header->msg_size = static_cast<uint16_t>(msg_size - 2);
+						header->msg_size = static_cast<uint16_t>(msg_size - 3);
```

Comment:
This is the Java code in jMAVlib: https://github.com/kd0aij/jMAVlib/blob/master/src/me/drton/jmavlib/log/ulog/ULogReader.java#L232